src-cpp/ProducerImpl.cpp

```diff
@@ -33,6 +33,16 @@
 
 #include "rdkafkacpp_int.h"
 
+const int RdKafka::Producer::RK_MSG_FREE;
+
+const int RdKafka::Producer::RK_MSG_COPY;
+
+const int RdKafka::Producer::RK_MSG_BLOCK;
+
+#ifndef MSG_COPY
```

Comment:
I think these always needs to be declared since the librdkafka and app compile defines may differ.
To be safe #undef MSG_COPY and MSG_FREE first here (but not in rdkafkacpp.h)